### ISSUE TITLE
Question about sage_conv in version 1.6.0

### ISSUE THREAD

**Issue by andewfitting** [pyg-team/pytorch_geometric#1496]
## ❓ Questions & Help

Sage_conv in master branch is quite different from original paper that have concatenation.

Could you explain more about   # propagate_type: (x: OptPairTensor)

 if isinstance(x, Tensor):
            x: OptPairTensor = (x, x)

 # propagate_type: (x: OptPairTensor)
 out = self.propagate(edge_index, x=x, size=size)

Thank You

**Comment by rusty1s** [comment]
Hi,

it is true that we do not use concatenation, but instead use summation for the addition of central node information. Note that both formulations are equivalent in case features are first transformed by a weight matrix.

Regarding your second point: You can use SAGEConv both for homogeneous and bipartite graphs. In case of the latter input features are given as a tuple of node features, where information from the source nodes x[0] is send to the destination nodes x[1]. Therefore, in case of homogeneous graphs, we can simply set x to (x, x).

**Comment by arunavsk** [comment]
@rusty1s any tips on what we might have to change to make it a concatenation? Also what does it mean to transform features by a weight matrix?

**Comment by rusty1s** [comment]
For concatenation, you can do the following:
```python
h = self.propagate(...)
h = torch.cat([x, h], dim=-1)
out = self.lin(h)
```
but it's actually the same as doing
```python
h = self.propagate(...)
out = self.lin1(x) + self.lin2(h)
```
The latter has the advantage of not explicitly creating a tensor of shape [num_nodes, 2 * num_features].
